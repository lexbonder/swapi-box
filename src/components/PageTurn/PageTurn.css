.PageTurn {
  width: fit-content;
  margin: auto;
}

.PageTurn button {
  background-color: transparent;
  border: 1px solid #feda4a;
  color: #4A53FE;
  font-size: 20px;
  outline: none;
  margin: 5px 30px;
}

.PageTurn button:hover {
  background-color: #feda4a;
}

.PageTurn button:active {
  background-color: #dcb929;
}