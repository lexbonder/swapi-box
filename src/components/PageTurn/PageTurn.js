import React from 'react';
import './PageTurn.css';
import PropTypes from 'prop-types';

const PageTurn = ({nextOrPrev}) => {
  // let lowest = 1;
  // let highest = 10;

  // const nextTen = (event) => {
  //   if (highest <= count) {
  //     nextOrPrev(event)
  //     lowest += 10;
  //     highest += 10;
  //   }
  //   if (highest > count) {
  //     highest = count;
  //   }
  // }

  // const lastTen = (event) => {
  //   nextOrPrev(event)
  //   lowest -= 10;
  //   highest -= 10;
  // }

  // <h2> {lowest} - {highest} of {count} </h2>
  return (
    <div className='PageTurn'>
      <button onClick={nextOrPrev} name='previous'>
          Last 10
      </button>
      <button onClick={nextOrPrev} name='next'>
          Next 10
      </button>
    </div>
  );
};

const { func } = PropTypes;

PageTurn.propTypes = {
  nextOrPrev: func
};

export default PageTurn;