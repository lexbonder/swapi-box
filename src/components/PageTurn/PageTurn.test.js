/* eslint-disable */
import React from 'react';
import {shallow} from 'enzyme';
import PageTurn from './PageTurn';

describe('PageTurn', () => {
  let wrapper;
  let mockNextOrPrev = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<PageTurn nextOrPrev={mockNextOrPrev} />)
  })

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call nextOrPrev when clicked', () => {
    wrapper.find('button').first().simulate('click')

    expect(mockNextOrPrev).toHaveBeenCalled();
  })

  it('should call nextOrPrev when clicked', () => {
    wrapper.find('button').last().simulate('click')

    expect(mockNextOrPrev).toHaveBeenCalled();
  })
})