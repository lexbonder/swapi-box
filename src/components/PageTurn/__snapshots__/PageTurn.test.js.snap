// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTurn Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageTurn
    nextOrPrev={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name="previous"
          onClick={[Function]}
>
          Last 10
</button>,
        <button
          name="next"
          onClick={[Function]}
>
          Next 10
</button>,
      ],
      "className": "PageTurn",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Last 10",
          "name": "previous",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Last 10",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next 10",
          "name": "next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next 10",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name="previous"
            onClick={[Function]}
>
            Last 10
</button>,
          <button
            name="next"
            onClick={[Function]}
>
            Next 10
</button>,
        ],
        "className": "PageTurn",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Last 10",
            "name": "previous",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Last 10",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next 10",
            "name": "next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next 10",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
